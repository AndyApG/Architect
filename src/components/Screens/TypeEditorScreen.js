import { compose, withProps } from 'recompose';
import TypeEditor, { formName } from '../TypeEditor';
import EditScreen from './EditScreen';

const typeEditorProps = withProps({
  editor: TypeEditor,
  form: formName,
});

const FormEditorScreen = compose(
  typeEditorProps,
)(EditScreen);

export default FormEditorScreen;
