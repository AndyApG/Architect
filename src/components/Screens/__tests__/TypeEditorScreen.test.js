/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import { TypeEditorScreen } from '../TypeEditorScreen';

const mockProps = {
  initialValues: {},
  onComplete: () => {},
  createType: () => {},
  updateType: () => {},
  submitForm: () => {},
};

describe('<TypeEditorScreen />', () => {
  it('can render', () => {
    const component = shallow(<TypeEditorScreen {...mockProps} />);

    expect(component).toMatchSnapshot();
  });
});
