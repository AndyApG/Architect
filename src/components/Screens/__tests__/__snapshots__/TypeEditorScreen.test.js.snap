// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TypeEditorScreen /> can render 1`] = `
<Connect(EditorScreen)
  editor={
    Object {
      "$$typeof": Symbol(react.memo),
      "WrappedComponent": [Function],
      "compare": null,
      "displayName": "Connect(withProps(withHandlers(withState(withHandlers(ReduxForm)))))",
      "type": [Function],
    }
  }
  form="TYPE_EDITOR"
/>
`;
